test/auth_test.rb

```diff
@@ -119,6 +119,55 @@ def app
         assert_equal "https://sso.heroku.com/initech",
                      last_response.headers["Location"]
       end
+
+      describe "with a valid token" do
+        let(:rack_env) do
+          {
+            "rack.session" => {
+              "sso_entity" => "initech",
+              "access_token" => "1234",
+              "access_token_expires_at" => (Time.now.getlocal + 3600).to_s
+            }
+          }
+        end
+
+        it "authorizes when i have a valid access token" do
+          stub_heroku_api do
+            get("/account") { status(200) }
+          end
+
+          post "/oauth/authorize", { client_id: "dashboard" }, rack_env
```

Comment:
Is `client_id` used by this endpoint? I don't see how it is when the request is a POST.